pkg/gitreceive/build.go

```diff
@@ -0,0 +1,297 @@
+	if err := yaml.Unmarshal(rawProcFile, &procType); err != nil {
+		return fmt.Errorf("procfile %s/ProcFile is malformed (%s)", tmpDir, err)
+	}
+
+	var srcManifest string
+	if err == os.ErrNotExist {
+		// both key and secret are missing, proceed with no credentials
+		if usingDockerfile {
+			srcManifest = "/etc/deis-dockerbuilder-no-creds.yaml"
+		} else {
+			srcManifest = "/etc/deis-slugbuilder-no-creds.yaml"
+		}
+	} else if err == nil {
+		// both key and secret are in place, so proceed with credentials
+		if usingDockerfile {
+			srcManifest = "/etc/deis-dockerbuilder.yaml"
+		} else {
+			srcManifest = "/etc/deis-slugbuilder.yaml"
+		}
+	} else if err != nil {
+		// unexpected error, fail
+		return fmt.Errorf("unexpected error (%s)", err)
+	}
+
+	fileBytes, err := ioutil.ReadFile(srcManifest)
+	if err != nil {
+		return fmt.Errorf("reading kubernetes manifest %s (%s)", srcManifest, err)
+	}
+
+	finalManifestFileLocation := fmt.Sprintf("/etc/%s", slugName)
+	var buildPodName string
+	var finalManifest string
+	uid := uuid.New()[:8]
+	if usingDockerfile {
+		buildPodName = fmt.Sprintf("dockerbuild-%s-%s-%s", appName, shortSha, uid)
+		finalManifest = strings.Replace(string(fileBytes), "repo_name", buildPodName, -1)
+		finalManifest = strings.Replace(finalManifest, "puturl", pushURL, -1)
+		finalManifest = strings.Replace(finalManifest, "tar-url", tarURL, -1)
+	} else {
+		buildPodName = fmt.Sprintf("slugbuild-%s-%s-%s", appName, shortSha, uid)
+		finalManifest = strings.Replace(string(fileBytes), "repo_name", buildPodName, -1)
+		finalManifest = strings.Replace(finalManifest, "puturl", pushURL, -1)
+		finalManifest = strings.Replace(finalManifest, "tar-url", tarURL, -1)
+		finalManifest = strings.Replace(finalManifest, "buildurl", buildPackURL, -1)
+	}
+
+	log.Debug("writing builder manifest to %s", finalManifestFileLocation)
+	if err := ioutil.WriteFile(finalManifestFileLocation, []byte(finalManifest), os.ModePerm); err != nil {
+		return fmt.Errorf("writing final manifest %s (%s)", finalManifestFileLocation, err)
+	}
+
+	configDir := "/var/minio-conf"
+	if err := os.MkdirAll(configDir, os.ModePerm); err != nil {
+		return fmt.Errorf("creating minio config file (%s)", err)
+	}
+
+	configCmd := mcCmd(configDir, "config", "host", "add", fmt.Sprintf("%s://%s:%s", storage.schema(), storage.host(), storage.port()), creds.key, creds.secret)
+	if err := run(configCmd); err != nil {
+		return fmt.Errorf("configuring the minio client (%s)", err)
+	}
+
+	makeBucketCmd := mcCmd(configDir, "mb", fmt.Sprintf("%s://%s:%s/git", storage.schema(), storage.host(), storage.port()))
+	// Don't look for errors here. Buckets may already exist
+	// https://github.com/deis/builder/issues/80 will eliminate this distaste
+	run(makeBucketCmd)
+
+	cpCmd := mcCmd(configDir, "cp", appTgz, tarURL)
+	cpCmd.Dir = repoDir
+	if err := run(cpCmd); err != nil {
+		return fmt.Errorf("copying %s to %s (%s)", appTgz, tarURL, err)
+	}
+
+	log.Info("Starting build")
```

Comment:
There's a long pause right here while we're waiting to boot the pod. Why not something like `Starting build... but first, coffee!` so people know that it's intentionally going to take a while